src/test/java/uk/nhs/hee/tis/trainee/sync/event/TrustEventListenerTest.java

```diff
@@ -96,4 +98,47 @@ void shouldSendRelatedPostsToQueueAfterSaveWhenRelatedPosts() {
     verify(messagingTemplate).convertAndSend(POST_QUEUE_URL, post3);
     assertThat("Unexpected table operation.", post3.getOperation(), is(Operation.LOAD));
   }
+
+  @Test
+  void shouldNotInteractWithPostQueueAfterDeleteWhenNoRelatedPosts() {
+    Document document = new Document();
+    document.append("_id", "trust1");
+    AfterDeleteEvent<Trust> event = new AfterDeleteEvent<>(document, Trust.class, "trust");
+
+    when(postService.findByEmployingBodyId("trust1")).thenReturn(Collections.emptySet());
+    when(postService.findByTrainingBodyId("trust1")).thenReturn(Collections.emptySet());
+
+    listener.onAfterDelete(event);
+
+    verifyNoInteractions(messagingTemplate);
+  }
+
+  @Test
+  void shouldSendRelatedPostsToQueueAfterDeleteWhenRelatedPosts() {
+    Document document = new Document();
+    document.append("_id", "trust1");
+
+    Post post1 = new Post();
+    post1.setTisId("post1");
+
+    Post post2 = new Post();
+    post2.setTisId("post2");
+
+    Post post3 = new Post();
+    post3.setTisId("post3");
+    when(postService.findByTrainingBodyId("trust1")).thenReturn(Set.of(post1, post2));
+    when(postService.findByEmployingBodyId("trust1")).thenReturn(Set.of(post2, post3));
```

Comment:
Noice.  Reliance on equality of `post2` from `postService.findByTrainingBodyId()` and `postService.findByEmployingBodyId()` but this was the coverage of "Set" behaviour I thought would be good to test.